src/diffusers/pipelines/free_noise_utils.py

```diff
@@ -30,6 +34,100 @@
+            **kwargs (`Dict[str, torch.Tensor]`):
+                Keyword arguments passed to the underlying `module`. Only keyword arguments whose names match the
+                entries in `input_kwargs_to_split` and are of type `torch.Tensor` will be split. The remaining keyword
+                arguments are passed unchanged.
+
+        Returns:
+            `Union[torch.Tensor, Tuple[torch.Tensor]]`:
+                The outputs obtained from `SplitInferenceModule` are the same as if the underlying module was inferred
+                without it.
+                - If the underlying module returns a single tensor, the result will be a single concatenated tensor
+                along the same `split_dim` after processing all splits.
+                - If the underlying module returns a tuple of tensors, each element of the tuple will be concatenated
+                along the `split_dim` across all splits, and the final result will be a tuple of concatenated tensors.
+
+        Workflow:
+            1. The keyword arguments specified in `input_kwargs_to_split` are split into smaller chunks using
+            `torch.split()` along the dimension `split_dim` and with a chunk size of `split_size`.
+            2. The `module` is invoked once for each split with both the split inputs and any unchanged arguments
+            that were passed.
+            3. The output tensors from each split are concatenated back together along `split_dim` before returning.
+
+        Example:
+            ```python
+            >>> import torch
+
+            >>> model = nn.Linear(1000, 1000)
+            >>> split_module = SplitInferenceModule(
+            ...     model, split_size=2, split_dim=0, input_kwargs_to_split=["input_data"]
+            ... )
+
+            >>> input_tensor = torch.randn(42, 1000)
+            >>> # Will split the tensor into 21 slices of shape [2, 1000].
+            >>> output = split_module(input_data=input_tensor)
+            ```
+
+        This method is useful when you need to perform inference on large tensors in a memory-efficient way by breaking
+        them into smaller chunks, processing each chunk separately, and then reassembling the results.
+
+        It is also possible to nest `SplitInferenceModule` across different split dimensions.
+        """
+        split_inputs = {}
+
+        # 1. Split inputs that were specified during initialization and also present in passed kwargs
+        for key in list(kwargs.keys()):
+            if key not in self.input_kwargs_to_split or not torch.is_tensor(kwargs[key]):
+                continue
+            split_inputs[key] = torch.split(kwargs[key], self.split_size, self.split_dim)
+            kwargs.pop(key)
+
+        # 2. Invoke forward pass across each split
+        results = []
+        for split_input in zip(*split_inputs.values()):
+            inputs = dict(zip(split_inputs.keys(), split_input))
+            inputs.update(kwargs)
```

Comment:
Clean 😎

Reply:
Not too proud of this one :grimacing: but hey, I'm not afraid of not understanding what sorcery is going on here two weeks later

![image](https://github.com/user-attachments/assets/7caa5dab-eddb-415e-86e6-5ce8fdc5990c)